web-app/django/VIM/apps/instruments/models/avresource.py

```diff
@@ -0,0 +1,41 @@
+from django.db import models
+
+
+class AVResource(models.Model):
+    RESOURCE_TYPE_CHOICES = [
+        ("audio", "Audio"),
+        ("video", "Video"),
+        ("image", "Image"),
+    ]
+
+    instrument = models.ForeignKey("Instrument", on_delete=models.PROTECT)
+    type = models.CharField(
+        max_length=5,
+        choices=RESOURCE_TYPE_CHOICES,
+        blank=False,
+        help_text="What type of audiovisual resource is this?",
+    )
+    format = models.CharField(
+        blank=False
+    )  # This should eventually be a choice field with supported formats
+    url = models.URLField(blank=False)
+    instrument_date = models.DateField(
+        blank=True, null=True, help_text="When was this instrument made?"
+    )
+    instrument_maker = models.CharField(
+        blank=True, help_text="Who made this instrument?"
+    )
+    # instrument_location; TBD how to manage location data
+    instrument_description = models.TextField(
+        blank=True, help_text="Additional information about the instrument."
+    )
+    instrument_description_language = models.ForeignKey(
+        "Language",
+        on_delete=models.PROTECT,
+        blank=True,
+        null=True,
+        help_text="What language is Instrument Description written in?",
+    )
```

Comment:
These fields are meant to describe the specific instance of the instrument seen/heard in the `AVResource`s?

This doesn't seem unreasonable, but strikes me as a bit odd since `AVResource`s are meant to be associated with (abstract) `Instrument`s. Is this metadata already attached to each of the `AVResource`s on WikiData, and we're creating fields to store it, just in case?

In summary, having these fields around probably won't hurt, but I don't see why they're necessary.

Reply:
> These fields are meant to describe the specific instance of the instrument seen/heard in the `AVResource`, yes?

Correct

> Is this metadata already attached to each of the `AVResource`s on WikiData, and we're creating fields to store it, just in case?

This metadata may be associated with a file that is shown on Wikidata. If so, it would actually be stored on the Wikimedia Commons. The main reason we're creating it is for an av resource added by a VIM user, to store details on the specific instance of the instrument contained therein.